File: gestor-biblioteca-web/src/app/modules/cadastro/livro/models/enums/categoria-livro.ts
export enum CategoriaLivro {
    ROMANCE = "Romance",
    FICCAO = "Ficção",
    NAO_FICCAO = "Não-ficção",
    FANTASIA = "Fantasia",
    CIENTIFICO = "Científico",
    BIOGRAFIA = "Biografia",
    HISTORIA = "História",
    AUTOAJUDA = "Autoajuda",
    MISTERIO = "Mistério",
    TERROR = "Terror",
    COMPUTADOR = "Computador"
}

export interface LabelValue {
    label: string;
    value: string;
}

export const CategoriaLivroLabelValue = [
    { label: 'Romance', value: 'ROMANCE' },
    { label: 'Ficção', value: 'FICCAO' },
    { label: 'Não-ficção', value: 'NAO_FICCAO' },
    { label: 'Fantasia', value: 'FANTASIA' },
    { label: 'Científico', value: 'CIENTIFICO' },
    { label: 'Biografia', value: 'BIOGRAFIA' },
    { label: 'História', value: 'HISTORIA' },
    { label: 'Autoajuda', value: 'AUTOAJUDA' },
    { label: 'Mistério', value: 'MISTERIO' },
    { label: 'Terror', value: 'TERROR' },
    { label: 'Computador', value: 'COMPUTADOR' }
  ];

File: gestor-biblioteca-api/src/main/java/br/hendrew/gestor_biblioteca/enums/CategoriaLivro.java
package br.hendrew.gestor_biblioteca.enums;

import br.hendrew.gestor_biblioteca.interfaces.Description;
import lombok.Getter;

@Getter
public enum CategoriaLivro implements Description {
    ROMANCE("Romance", "Romance"),
    FICCAO("Ficção", "Fiction"),
    NAO_FICCAO("Não-ficção", "Non-fiction"),
    FANTASIA("Fantasia", "Fantasy"),
    CIENTIFICO("Científico", "Science"),
    BIOGRAFIA("Biografia", "Biography"),
    HISTORIA("História", "History"),
    AUTOAJUDA("Autoajuda", "Self-help"),
    MISTERIO("Mistério", "Mystery"),
    TERROR("Terror", "Horror"),
    COMPUTADOR("Computador", "Computers"),
    EDUCACAO("Educação","Education"),
    OUTRAS("Outras", "Outher");

    private final String description;
    private final String descriptionEn;

    CategoriaLivro(String description, String descriptionEn) {
        this.description = description;
        this.descriptionEn = descriptionEn;
    }

    public static CategoriaLivro fromCategoria(String descricaoEn) {
        for (CategoriaLivro categoria : CategoriaLivro.values()) {
            if (categoria.getDescriptionEn().equalsIgnoreCase(descricaoEn)) {
                return categoria;
            }
        }
        return OUTRAS;
    }
}
